src/Centreon/Application/Controller/DowntimeController.php

```diff
@@ -0,0 +1,475 @@
+        $contact = $this->getUser();
+        if ($contact === null) {
+            return $this->view(null, Response::HTTP_UNAUTHORIZED);
+        }
+        $this->monitoringService->filterByContact($contact);
+
+        if ($this->monitoringService->isHostExists($hostId)) {
+            $downtimesByHost = $this->downtimeService
+                ->filterByContact($contact)
+                ->findDowntimesByService($hostId, $serviceId);
+
+            $context = (new Context())->setGroups(['dwt_main', 'dwt_service']);
+
+            return $this->view([
+                'result' => $downtimesByHost,
+                'meta' => [
+                    'pagination' => $requestParameters->toArray()
+                ]
+            ])->setContext($context);
+        } else {
+            return View::create(null, Response::HTTP_NOT_FOUND, []);
+        }
+    }
+
+    /**
+     * Entry point to find one host downtime.
+     *
+     * @IsGranted("ROLE_API_REALTIME", message="You are not authorized to access this resource")
+     * @Rest\Get(
+     *     "/monitoring/downtimes/{downtimeId}",
+     *     requirements={"downtimeId"="\d+"},
+     *     condition="request.attributes.get('version.is_beta') == true",
+     *     name="monitoring.downtime.findOneDowntime")
+     * @param int $downtimeId Downtime id to find
+     * @return View
+     * @throws \Exception
+     */
+    public function findOneDowntime(int $downtimeId): View
+    {
+        $contact = $this->getUser();
+        if ($contact === null) {
+            return $this->view(null, Response::HTTP_UNAUTHORIZED);
+        }
+        $downtime = $this->downtimeService
+            ->filterByContact($contact)
+            ->findOneDowntime($downtimeId);
+
+        if ($downtime !== null) {
+            $context = (new Context())
+                ->setGroups(['dwt_main', 'dwt_service'])
+                ->enableMaxDepth();
+
+            return $this->view($downtime)->setContext($context);
+        } else {
+            return View::create(null, Response::HTTP_NOT_FOUND, []);
+        }
+    }
+
+    /**
+     * Entry point to find the last downtimes.
+     *
+     * @IsGranted("ROLE_API_REALTIME", message="You are not authorized to access this resource")
+     * @Rest\Get(
+     *     "/monitoring/downtimes",
+     *     condition="request.attributes.get('version.is_beta') == true",
+     *     name="monitoring.downtime.findDowntimes")
+     * @param RequestParametersInterface $requestParameters
+     * @return View
+     * @throws \Exception
+     */
+    public function findDowntimes(RequestParametersInterface $requestParameters): View
+    {
+        $contact = $this->getUser();
+        if ($contact === null) {
+            return $this->view(null, Response::HTTP_UNAUTHORIZED);
+        }
+        $hostsDowntime = $this->downtimeService
+            ->filterByContact($contact)
+            ->findDowntimes();
+
+        $context = (new Context())->setGroups(['dwt_main', 'dwt_service']);
```

Comment:
should not be better to use complete name ? downtime_main and downtime_service

Reply:
Why not